app/src/components/datasets/DatasetContentTabs/Evaluation/AddEvalModal.tsx

```diff
@@ -231,17 +233,23 @@ const AddEvalModal = ({ disclosure }: { disclosure: UseDisclosureReturn }) => {
             <Button colorScheme="gray" onClick={disclosure.onClose} minW={24}>
               Cancel
             </Button>
-            <Button
-              colorScheme="blue"
-              onClick={onCreationConfirm}
-              minW={24}
-              isLoading={creationInProgress}
-              isDisabled={
-                !name || !instructions || !numDatasetEntries || includedModelIds.length < 2
-              }
-            >
-              Create
-            </Button>
+            <AccessControl accessLevel="requireCanModifyProject">
+              <Button
+                colorScheme="blue"
+                onClick={onCreationConfirm}
+                minW={24}
+                isLoading={creationInProgress}
+                isDisabled={
+                  !name ||
+                  !instructions ||
+                  !numDatasetEntries ||
+                  includedModelIds.length < 2 ||
+                  insufficientPermission
```

Comment:
Unfortunately, when buttons possess their own `isDisabled` prop, AccessControl doesn't override it, so we need to manually disable them. When buttons don't define `isDisabled` for themselves, AccessControl automatically disables them when appropriate.

Reply:
Just thought of another possible abstraction here that will actually lead to a better UX and also solve this problem more cleanly. What if we generalized the access control wrapper component into something like this:

```ts
<ConditionallyEnable
  accessRequired="requireCanViewProject"
  checks={[
    [selectedRows > 10, "You must select at least 10 rows before starting a fine-tune"],
    [modelName === 'gpt-3.5' || hasBetaAccess, "You need beta access before you can fine-tune this model"]
    // ...etc
  ]}
>
// ...
```

Basically, the idea is to move all the `isDisabled` checks into the higher-level access control wrapper. That way we can also easily add a tooltip for each reason we might disable the button, which will provide a better UX.